### PR TITLE
Fix RoundNearestTiesAway

### LINKED ISSUE #29698: RoundNearestTiesUp is incorrect for small negative values
```
julia> round(-1e-80, RoundNearestTiesUp)
-1.0
```

### PR DESCRIPTION
**Pull request by simonbyrne**
Fixes #29698. Idea thanks to @raphlinus (https://twitter.com/raphlinus/status/1052646958962159616).

### CODE REVIEW

**Comment by simonbyrne** [comment]
Note that this won't set the floating point status flags correctly.

File: base/floatfuncs.jl [hunk 1]
```diff
@@ -214,9 +214,8 @@ function round(x::AbstractFloat, ::RoundingMode{:NearestTiesAway})
     ifelse(x==y,y,trunc(2*x-y))
 end
 # Java-style round
-function round(x::AbstractFloat, ::RoundingMode{:NearestTiesUp})
-    y = floor(x)
-    ifelse(x==y,y,copysign(floor(2*x-y),x))
+function round(x::T, ::RoundingMode{:NearestTiesUp}) where {T <: AbstractFloat}
+    copysign(floor((x + (T(0.25) - eps(T(0.5)))) + (T(0.25) + eps(T(0.5)))), x)
```
Comment (StefanKarpinski) [inline base/floatfuncs.jl]: Would `prevfloat` and `nextfloat` be clearer here?

File: test/rounding.jl [hunk 2]
```diff
@@ -134,6 +134,15 @@ end
...
     end
 
+    for Tf in [Float16,Float32,Float64]
+        @test round(eps(Tf)/8, RoundNearest) === zero(Tf)
+        @test round(-eps(Tf)/8, RoundNearest) === -zero(Tf)
+        @test round(eps(Tf)/8, RoundNearestTiesAway) === zero(Tf)
+        @test round(-eps(Tf)/8, RoundNearestTiesAway) === -zero(Tf)
+        @test round(eps(Tf)/8, RoundNearestTiesUp) === zero(Tf)
+        @test round(-eps(Tf)/8, RoundNearestTiesUp) === -zero(Tf)
+    end
```
Comment (StefanKarpinski) [inline test/rounding.jl]: What about testing each rounding behavior at the high and low value of each possible value of the exponent? That should be reasonably quick and fairly thorough. I know that it's probably overkill but, why not?

**Comment by StefanKarpinski** [comment]
Backportable?

**Pull request merged**